Simple-PHP-IPAM/lib.php

```
@@ -3281,3 +3282,289 @@ function der_len(int $len): string
+/**
+ * Probe a single IP via TCP connection.
+ *
+ * The IP MUST be validated by normalize_ip() before this call is made.
+ *
+ * @return int|null  Connection latency in milliseconds, or null on failure.
+ */
+function ipam_probe_tcp(string $ip, int $port, int $timeoutMs = 1000): ?int
+{
+    $timeout = $timeoutMs / 1000.0;
+    $start = microtime(true);
+    $sock = @fsockopen($ip, $port, $errno, $errstr, $timeout);
```

PHPMD (2.15.0)

[warning] Avoid unused local variables such as '$errno'. (undefined)

(UnusedLocalVariable)